### ISSUE TITLE
SSL redirect loop happening within ajax requests for messages blocks

### ISSUE THREAD

**Issue by jessek1** [nexcess/magento-turpentine#453]
**Current setup and versions**

Ubuntu Server 13.10
Magento 1.8.1
Turpentine version feature better 1.8 compat
Nginx 1.4.1 listening on 443 reverse proxying to 127.0.0.1:80
Varnish 3.0.4 listening on 80 reverse proxying to 127.0.0.1:8080
Apache 2.4.6 listening on 8080

**Issue**
Everything seems to work fine using non-SSL, but some SSL pages end up in redirect loop within ajax requests.

Going to the login page causes a strange loop of ajax requests that seem to keep reloading the content of the page within each other, but the actual page is not redirecting itself.

In the below trace, you'll see that the ESI blocks keep redirecting to each other, this continues forever, however in the trace I only show the first and beginning of the second loop.

Other SSL pages such as the Magento admin area operate as expected.

**Req/Resp trace**

Full trace : http://pastebin.com/BtMpkVSm
Only headers trace : http://pastebin.com/UxsuMhhQ

**System.log from magento**
http://pastebin.com/MF1b2fMw

Any help would be much appreciated.

**Update 1 3/6/14 11:52AM PST**
I've been able to get around the redirects by disabling the cache in the ESI policy for the following blocks.  

To do this I had to comment out the default contents and add the `<turpentine_cache_flag value="0"/>` node.

After refreshing the caches, the redirect were no longer happening.

Understandably there will be a performance impact as I've now disabled the cache for the messages blocks, but what are these for and what would be the *real* fix?

```
 <reference name="global_messages">
            <turpentine_cache_flag value="0"/>
            <!--
            <action method="setEsiOptions">
                <params>
                    <access>private</access>
                    <method>ajax</method>
                    <-- if this is ever cached we should flush on this event
                    <flush_events>
                        <core_session_abstract_add_message/>
                    </flush_events>
                    ->
                </params>
            </action>
            -->
        </reference>
        <reference name="messages">
            <turpentine_cache_flag value="0"/>
            <!--
            <action method="setEsiOptions">
                <params>
                    <access>private</access>
                    <method>ajax</method>
                    <!-- if this is ever cached we should flush on this event
                    <flush_events>
                        <core_session_abstract_add_message/>
                    </flush_events>
                    ->
                </params>
            </action>
            -->
        </reference>
```

**Comment by q23** [comment]
running into same problem. her is the solution:
https://github.com/eth8505/magento-turpentine/commit/575f499382217f0013eaf097fd79ceddec0b4381

**Comment by miguelbalparda** [comment]
This should be fixed by #844.
